### ISSUE TITLE
When refiling: Invalid function: org-preserve-local-variables

### ISSUE THREAD

**Issue by alexlai97** [hlissner/doom-emacs#3649]
**What did you expect to happen?**

refile correctly with no error when triggerring `org-refile`

**What actually happened?**
I updated (and upgraded) doom emacs today. I encountered the same from a previous [issue](https://github.com/hlissner/doom-emacs/issues/1116). I tried 
- `doom compile :plugins`
- `doom purge && doom compile`
- `doom build`
-  reinstall doom emacs by deleting `~/.emacs.d` and git clone and `doom install`

However, the problem remains. 

When I call `org-refile`, \*Backtrace\*
```
Debugger entered--Lisp error: (invalid-function org-preserve-local-variables)
  org-preserve-local-variables("Copied: Subtree(s) with 1190 characters")
  org-copy-subtree(1 nil t)
  org-refile(nil)
  funcall-interactively(org-refile nil)
  call-interactively(org-refile nil nil)
  command-execute(org-refile)
```
in \*Messages\*
```
Copied: Subtree(s) with 1190 characters
org-copy-subtree: Invalid function: org-preserve-local-variables
```

**Steps to reproduce:**
`M-x org-refile`

**System information:**
<details><pre>
output of `M-x doom/info`:
```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      May 19, 2020
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-wide-int --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      grafted, HEAD -> develop, origin/develop, origin/HEAD 96ab60d 2020-07-29 16:10:31 -0400
        dir        ~/Projects/myrepos/dotfiles/doom-emacs/.config/doom/
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/fish
        uname      Linux 5.4.53-1-lts #1 SMP Wed, 22 Jul 2020 20:16:09 +0000 x86_64
        path       (~/.scripts ~/go/bin ~/.local/share/cargo/bin ~/.luarocks/bin ~/.emacs.d/bin ~/.config/vifm/scripts ~/.local/bin /opt/bin /usr/local/bin /usr/bin /sbin/ /bin /usr/sbin/ /usr/bin/core_perl/ ~/.local/share/texlive/2020/bin/x86_64-linux/ /usr/lib/emacs/26.3/x86_64-pc-linux-gnu/)
config  envfile    envvar-file
        elc-files  186
        modules    (:completion company ivy :ui deft doom doom-dashboard doom-quit fill-column hl-todo modeline nav-flash neotree ophints (popup +defaults) vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) file-templates fold multiple-cursors snippets :emacs (dired +icons +ranger) ibuffer vc :term eshell term :checkers syntax :tools debugger (eval +overlay) gist lookup lsp magit pdf rgb tmux :lang cc data emacs-lisp (go +lsp) (haskell +dante) javascript latex lua markdown (org +roam +dragndrop +gnuplot +hugo +pandoc +present) (python +lsp) (rust +lsp) (sh +fish +lsp) web :email notmuch :app calendar :config literate (default +bindings +smartparens))
        packages   ((simplenote2) (emojify) (org-roam-server))
        unpin      (n/a)
        elpa       (simple-httpd s org-roam-server org-roam org-re-reveal org htmlize f emacsql-sqlite3 emacsql dash)
```
</pre></details>

**Comment by hlissner** [comment]
You have a conflict between a) the packages you have installed with package.el (e.g. `M-x package-install` or `use-package`'s :ensure keyword):

```
        packages   ((simplenote2) (emojify) (org-roam-server))
        unpin      (n/a)
        elpa       (simple-httpd s org-roam-server org-roam org-re-reveal org htmlize f emacsql-sqlite3 emacsql dash)
```

1. Stop using package.el (remove all package-* calls in your config and `:ensure t`'s in your `use-package` blocks, if any).
2. Run `doom sync -p`. `-p` will **p**urge any package.el-installed packages on your system.
3. Restart Emacs.

You don't need to install any of the above packages yourself; they are all installed for you by the Doom modules you already have enabled.

**Comment by polaris64** [comment]
I am also having this problem since this morning's upgrade. I have checked the above and all of my `use-package` calls are not using `:ensure t` and I don't have any `package-*` calls as far as I can tell. I have also run `doom sync -p` but the problem still persists.

Running `doom doctor` shows the following message: "‘org’ was already loaded by the time lang/org loaded, this may cause issues".  I'm guessing this is related?

I'm using some org-related packages in my `packages.el` file as follows: -
```
(package! ox-twbs)
(package! org-drill)
```

Hopefully this helps to narrow down the problem.

**Comment by hlissner** [comment]
I'm surprised `doom build` didn't fix your issue, in that case do you have a link to your config?

**Comment by alexlai97** [comment]
> With the exception of simplenote2 and emojify, you don't need to install any of the above packages yourself; they are all installed for you by the Doom modules you already have enabled.

This is my current `package.el`
```
(package! simplenote2)
(package! emojify)
```
Do you mean that I can load them using doom modules ? I thought i have to install them using `package!` since i didn't find related modules in the this doom-emacs repo. What's the correct way of doing this ?


> 1.  Stop using package.el (remove all package-* calls in your config and :ensure t's in your use-package blocks, if any).
> 2. Run doom sync -p. -p will purge any package.el-installed packages on your system.
> 3.  Restart Emacs.

What i have tried now is:
1. comment out `org-roam-sever` (that's introducing `org` dependency) in `packages.el`
```
;;; my current packages.el
(package! simplenote2)
(package! emojify)
;; (package! org-roam-server)
```
2. comment out org-roam-server related config in `use-package` block.
3. doom sync -p , restart emacs

However, the problem remains. `(invalid-function org-preserve-local-variables)`

Another question is:
- how to install org-roam-server correctly if not using `package!` in `packages.el` ?

**Comment by brabalan** [comment]
I have the same issue. Here is the relevant `doom/info` part:
```
        modules    (:completion (company +childframe) (ivy +icons) :ui deft doom doom-dashboard doom-quit hl-todo modeline ophints (popup +defaults) vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) file-templates fold multiple-cursors snippets :emacs (dired +icons) electric undo vc :term eshell :checkers syntax spell :tools (eval +overlay) lookup magit make pass pdf :lang coq data emacs-lisp javascript latex ledger lua markdown ocaml (org +gnuplot +roam +journal +noter) sh :email notmuch :app calendar irc (rss +org) :config literate (default +bindings +smartparens))
        packages   ((ebdb) (emojify) (org-review) (xah-replace-pairs) (org-ref) (gnuplot) (ts) (magit-todos :disable t) (elpher) (org-roam-bibtex) (leuven-theme) (unicode-fonts) (beacon) (xah-math-input))
        unpin      (n/a)
        elpa       (n/a)
```

**Comment by hlissner** [comment]
Is it possible that any of you have elisp packages installed through your OS package manager? For instance, mu (for mu4e). Some OSes may be loading org long before `~/.emacs.d` is sourced, meaning you are forever stuck with the old org that is built into Emacs. That was true for gentoo users for a time, it may be for other OS users. You'd have to check your site files.

In any case, the problem is that org is being loaded to early _by something_, and I suspect that something, if not in your config (in your init.el's in particular), then by something outside of it, which I have no control over.

I also notice that @alexlai97's config is byte-compiled. Please run `doom clean`, then try `doom sync && doom build` and see if that changes anything. It's never a good idea to debug issues while your config is byte-compiled.

**Comment by polaris64** [comment]
Here's some more information from my perspective, hopefully this will help.

I've just tried this on my second computer and everything worked fine, and `doom doctor` reported no issues. I upgraded Doom (`doom upgrade`) with these changes: https://github.com/hlissner/doom-emacs/compare/aac73f4c01ed6515347dc6b264c20230d7e8bb71...e07cc77c19a410bc38bf9a109ad01983a123ecd2. During this upgrade process, `org-mode` updated from 82b496c -> a1e5bee. Following this upgrade `doom doctor` reports the above issue again and refiling no longer works. So something must have happened between these commits to cause this.

Henrik, to answer your previous comment directly, I have only the following Emacs-related packages installed on my system: -

```
emacs-bin-common/focal,now 1:26.3+1-1ubuntu2 amd64 [installed,automatic]
emacs-common-non-dfsg/focal,focal,now 1:26.3+1-1 all [installed]
emacs-common/focal,focal,now 1:26.3+1-1ubuntu2 all [installed,automatic]
emacs-el/focal,focal,now 1:26.3+1-1ubuntu2 all [installed,automatic]
emacs-gtk/focal,now 1:26.3+1-1ubuntu2 amd64 [installed]
emacs/focal,focal,now 1:26.3+1-1ubuntu2 all [installed]
emacsen-common/focal,focal,now 3.0.4 all [installed,automatic]
```

I have `mailbox-utils` installed but not the `mu4e` system (Ubuntu) package. I do have `mu4e` enabled in `init.el`.

The value of `org-version` when running Doom Emacs is "9.4" and when running `emacs -Q` it is "9.1.9".

**Comment by alexlai97** [comment]
> I also notice that @alexlai97's config is byte-compiled. Please run `doom clean`, then try `doom sync && doom build` and see if that changes anything. It's never a good idea to debug issues while your config is byte-compiled.
No, nothing's changed.
> In any case, the problem is that org is being loaded to early by something, and I suspect that something, if not in your config (in your init.el's in particular), then by something outside of it, which I have no control over.
Yeah, i think that's case as well. Need to dig into where causes org to load early when i have time.

**Comment by polaris64** [comment]
I've been trying to figure out what's going on and I have a feeling it might be something to do with mu4e loading Org too early as you suggested. However I've removed mu4e completely from my system, from `init.el`, `packages.el` and my custom `config.el`. In fact I removed *all* custom config. I then ran `doom sync -p`, `doom clean` and `doom build` and I still had the same problem (and the same message from `doom doctor`) so I'm not sure if mu4e is actually the problem.

Nevertheless, I wanted to ask if my set-up is correct.  I'm using Ubuntu 20.04 and I have the `maildir-utils` system package installed as described in the mu4e module readme. If I un-comment `(mu4e +gmail)` in `init.el` I get the "No such file or directory, mu4e" error [mentioned here](https://github.com/hlissner/doom-emacs/tree/develop/modules/email/mu4e#troubleshooting) when trying to run `mu4e`. The only way I have been able to get mu4e working properly is to install the `mu4e` system package, which installs mu4e .el files in `/usr/share/emacs/site-packages/`. Installing this, then running `doom sync` again allows mu4e to operate properly.

Is this the correct way of doing it however? I assumed that `(mu4e +gmail)` in `init.el` would install mu4e itself and the only support package required on the system would be `maildir-utils` which provides `mu`. But it seems that this is not the case and I also need to install the `mu4e` Ubuntu package.

Could you confirm whether I'm doing this correctly? I have a feeling this is something to do with this problem.

Thanks!

**Comment by alexlai97** [comment]
I updated doom emacs today and the issue was gone. 😇

**Comment by hlissner** [comment]
Ah, my bad. I fixed this a short while ago, in f3740d476, and forgot to tag or mention it! If the issue resurfaces let me know.

**Comment by jonathan-sahar-tau** [comment]
Hi, I upgraded Doom a couple of days ago and I'm experiencing the same.
